pushnotifications/src/main/java/com/pusher/pushnotifications/internal/PersistentJobQueue.kt

```diff
@@ -12,67 +12,26 @@ interface PersistentJobQueue<T: Serializable> {
   fun asIterable(): Iterable<T>
 }
 
-class TapeJobQueue<T: Serializable>(file: File): PersistentJobQueue<T> {
-  private val queueFile = QueueFile.Builder(file).build()
-  private val log = Logger.get(this::class)
+class TapeJobQueue<T: Serializable>(file: File, converter: ObjectQueue.Converter<T>): PersistentJobQueue<T> {
+  private var queue = ObjectQueue.create( QueueFile.Builder(file).build(), converter )
 
-  override fun push(job: T) {
-    val byteOutputStream = ByteArrayOutputStream()
-    val objectOutputStream = ObjectOutputStream(byteOutputStream)
-    objectOutputStream.writeObject(job)
-    objectOutputStream.flush()
-    val jobBytes = byteOutputStream.toByteArray()
-
-    synchronized(queueFile) {
-      queueFile.add(jobBytes)
-    }
+  @Synchronized override fun push(job: T) {
+    queue.add(job)
   }
 
-  override fun peek(): T? {
-    val jobBytes = synchronized(queueFile) {
-      queueFile.peek() ?: return null
-    }
-
-    val byteInputStream = ByteArrayInputStream(jobBytes)
-    val objectInputStream = ObjectInputStream(byteInputStream)
-
-     try {
-      @Suppress("unchecked_cast")
-      return objectInputStream.readObject() as T
-    } catch (e: InvalidClassException) {
-       log.w("Failed to read data from tape. Continuing without this data.")
-    }
-
-    return null
+  @Synchronized override fun peek(): T? {
+    return queue.peek()
```

Comment:
What will happen if the data is corrupted? It seems like we no longer handle this here (at least explicitly).